pydantic_xml/fields.py

```diff
@@ -292,36 +292,36 @@ def computed_element(
     return computed_entity(EntityLocation.ELEMENT, prop, path=tag, ns=ns, nsmap=nsmap, nillable=nillable, **kwargs)
 
 
-ValidatorFuncT = TypeVar('ValidatorFuncT', bound='model.SerializerFunc')
-
-
-def xml_field_validator(field: str, /, *fields: str) -> Callable[[ValidatorFuncT], ValidatorFuncT]:
+def xml_field_validator(
+    field: str, /, *fields: str
+) -> 'Callable[[model.ValidatorFuncT[model.ModelT]], model.ValidatorFuncT[model.ModelT]]':
     """
     Marks the method as a field xml validator.
 
     :param field: field to be validated
     :param fields: fields to be validated
     """
 
-    def wrapper(func: ValidatorFuncT) -> ValidatorFuncT:
+    def wrapper(func: model.ValidatorFuncT[model.ModelT]) -> model.ValidatorFuncT[model.ModelT]:
+        if isinstance(func, (classmethod, staticmethod)):
```

Comment:
Could you explain what is the purpose of the check for staticmethod? It seems to me that a validator method may not be static.

Reply:
@dapper91 my thinking is that it's valid to declare a validator both as
```python
    @xml_field_validator('x', 'y')
    @classmethod
    def validate_space_separated_list(cls, element: XmlElementReader, field_name: str) -> List[float]:
```
and as
```python
    @xml_field_validator('x', 'y')
    @staticmethod
    def validate_space_separated_list(cls: type[BaseXmlModel], element: XmlElementReader, field_name: str) -> List[float]:
```
mypy accepts both, so I decided to support both.

That said, I don't know of a good reason to use the latter form. (Maybe it can happen when trying to reuse code with the newly added `XmlFieldValidator`?) I can remove the check for staticmethod if you prefer.